frontend/packages/knative-plugin/src/components/test-function/modal-components/RequestOptions.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { Button, TextInputTypes } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { FormikProps, FormikValues } from 'formik/dist/types';
+import { useTranslation } from 'react-i18next';
+import { AsyncComponent } from '@console/internal/components/utils/async';
+import { DropdownField, InputField } from '@console/shared/src/components';
+import { FormLayout } from '@console/shared/src/components/cluster-configuration';
+import { InvokeFormat } from '../../../types';
+import '../TestFunctionModal.scss';
+
+const NameValueEditorComponent = (props) => (
+  <AsyncComponent
+    loader={() =>
+      import('@console/internal/components/utils/name-value-editor').then((c) => c.NameValueEditor)
+    }
+    {...props}
+  />
+);
+
+const RequestOptions: React.FC<FormikProps<FormikValues>> = ({ setFieldValue, values }) => {
+  const { t } = useTranslation();
+  const [showCustomHeaders, setShowCustomHeaders] = React.useState(false);
+  const {
+    request: {
+      headers: {
+        [InvokeFormat.HTTP]: { custom: customHTTPHeaders },
+        [InvokeFormat.CloudEvent]: { custom: customCloudEventHeaders },
+      },
+      format: invokeFormat,
+    },
+  } = values;
+
+  const handleNameValuePairs = ({ nameValuePairs: updatedNameValuePairs }) => {
+    /* Removing the extra element from arrays that are auto-generated */
+    updatedNameValuePairs.forEach((arr) => arr.length >= 3 && arr.splice(2, 1));
+
+    setFieldValue(`request.headers.${invokeFormat}.custom`, updatedNameValuePairs);
+  };
+
+  const checkCustomHeadersIsEmpty = () => {
+    let filteredArray = [];
+    if (invokeFormat === InvokeFormat.HTTP) {
+      filteredArray = customHTTPHeaders.filter((arr) => !arr.every((element) => element === ''));
+    } else if (invokeFormat === InvokeFormat.CloudEvent) {
+      filteredArray = customCloudEventHeaders.filter(
+        (arr) => !arr.every((element) => element === ''),
+      );
+    }
+    return filteredArray.length === 0;
+  };
+
+  return (
+    <>
+      <div className="kn-test-sf-modal-request-pane--options__size">
+        <FormLayout isHorizontal>
+          <DropdownField
+            label="Format"
+            name="request.format"
+            items={InvokeFormat}
```

Comment:
Create a small items object that makes the values translatable.

```ts
const formatItems = {
  [InvokeFormat.CloudEvent]: t('knative-plugin~CloudEvent'),
  [InvokeFormat.HTTP]: t('knative-plugin~HTTP'),
}
```

```suggestion
            items={formatItems}
```